### PR TITLE
More Python tests for materialized view and Alternator GSI feature

### PR DESCRIPTION
**Pull request by nyh**
This patch includes more tests (in Python) that I wrote while implementing the Alternator UpdateTable feature for adding a GSI to an existing table (https://github.com/scylladb/scylladb/issues/11567).

I explain each of these tests in the separate patches below, but basically they fall into two types:
1. Tests which pass with today's materialized views and Alternator GSI/LSI, and serve to ensure that whatever changes I do to the view update implementation, doesn't break corner cases that already worked.
2. Tests for the UpdateTable feature in Alternator which doesn't work today so xfail - and will need to work for #11567. We already had a few tests for this, but here I add more and improve coverage of various corner cases I discovered while implementing the featue.

I already have a working prototype for #11567 which passes all these tests. Many of these tests helped exposed various bugs in earlier versions of my code.

### CODE REVIEW

File: test/alternator/test_gsi.py
Comment (piodul) [inline test/alternator/test_gsi.py:10]: > This patch adds many more tests for various corner cases of
> this feature - tests developed in parallel with actually implementing
> that feature.
> 
> The source file test/alternator/test_gsi.py has already grown pretty
> large, so this patch moves all the tests related to using UpdateTable
> to add or delete a GSI - both the existing tests and the new ones -
> to a separate file: test_gsi_updatetable.py,

Please separate the code movement and addition of new tests into two commits.

**Review by piodul — COMMENTED** [review]
Only one comment, otherwise LGTM.

I am not too familiar with Alternator, so somebody else more familiar with it should review as well - @ptrsmrn could you or somebody from your team take a look?

File: test/alternator/test_gsi.py
Comment (nyh) [reply test/alternator/test_gsi.py:10]: I can do this, but it won't be trivial, as I actually made some changes in comments and perhaps (I don't remember) other small changes in the code after the movement. 

From the **deletion** side you can see exactly what I moved (which got deleted from the old file), but you're right that you can't see exactly what I changed after the move.

**Comment by nyh** [comment]
CI failure is a flaky test - https://github.com/scylladb/scylladb/issues/21872

File: test/alternator/test_cql_rbac.py
Comment (ptrsmrn) [inline test/alternator/test_cql_rbac.py:675]: "Above" - that is, unless sombody doesn't insert a testcase in between.

File: test/alternator/test_cql_rbac.py
Comment (ptrsmrn) [inline test/alternator/test_cql_rbac.py:677]: So creating a table already with the GSI is not possible and we first need to create a table and then add a GSI to it via ALTER?

Comment (nyh) [reply test/alternator/test_cql_rbac.py:677]: No - we already supported since the first days of Alternator the `CreateTable` API to create a table with a GSI, what we're adding now is the `UpdateTable` API to modify an existing table by adding a GSI to it.
The `CreateTable` requires (unsurprisingly) the CREATE permissions, but also auto-grants all the permissions - including ALTER - to the role who was allowed to create the table, so this role is later able to add another GSI to this table if he wishes.

File: test/alternator/test_cql_rbac.py
Comment (nyh) [reply test/alternator/test_cql_rbac.py:675]: Yes, I'm writing a book here :-)
Also above here doesn't mean the immediately above test, just that we had another test for it. When I want to refer to a specific test I usually include its name exactly for the reason you asked about. I can do it here too.

Comment (nyh) [reply test/alternator/test_cql_rbac.py:675]: Fixing the text to "Above in test_rbac_updatetable() we checked". 

I know nothing guarantees that it will remain "above", but I think there's a high likelyhood that nobody will ever come to this test file and shuffle around functions randomly, so the word "above" is likely to remain correct (and if not, it can be fixed when becomes incorrected).

Comment (nyh) [reply test/alternator/test_cql_rbac.py:675]: I'm trying to do this but lacking the git skills, apparently :-( I'm still thinking how to do it without losing my mind.

Comment (nyh) [reply test/alternator/test_cql_rbac.py:675]: I managed to do it :-) Mostly using "git rebase -i" as the tool.

**Comment by nyh** [comment]
Pushed new version with:
1. Fixed one test that I found was flaky - test_rbac_updatetable_gsi() - the test for permission denied on table **delete** needs to be a bit more clever - it needs retry but we can't retry deletion because if the first delete succeeds and deletes the GSI - we can't continue the iterations.
2. Fixed one comment as requested by a reviewer.
3. Instead of moving existing UpdateTable tests, improving them and adding more - all in one patch - now I split this change into three patches (requested by a reviewer).

**Comment by nyh** [comment]
Please review again and/or consider merging. I believe I implemented the reviewers' comments, and the CI passed.

**Review by ptrsmrn — APPROVED** [review]

**Comment by nyh** [comment]
@scylladb/scylla-maint please consider merging this pull request. I am waiting for it to be merged to push on top of it the actual implementation of that GSI feature.

**Review by piodul — APPROVED** [review]

**Comment by piodul** [comment]
Queued

**Pull request merged**
